### DISCUSSION TITLE
Sofa.ConfigConfig.cmake vs. SofaConfig.cmake

### DISCUSSION THREAD

**Discussion by dennis2society** [sofa-framework/sofa#4175] [Build / Config / Environment] [answered]
Hi,
I am trying to use some of the Sofa libs  in my own project but I am running in an issue with the SofaConfig.cmake:

tldr;
Using `find_package(Sofa)` in my own CMakeLists.txt is looking for `SofaConfig.cmake` which does not exist. Instead there is a `Sofa.ConfigConfig.cmake` in `$sofa/build/install/lib/cmake/Sofa.Config/`. Copying this file and renaming it to `SofaConfig.cmake` solves the problem. But what's the issue here? Is my CMakelists wrong? Or should `Sofa.ConfigConfig.cmake` be renamed to `SofaConfig.cmake`?

##Long version:

I have a fresh clone of the git repo and built Sofa from source:

`cmake -B build/ -D CMAKE_BUILD_TYPE=Release`
`cmake --build build/`
`cmake --install build/`

Everything went fine and I have an install folder inside by build folder. 

To use Sofa in my own project, I have written a CMakeLists.txt:

```
  cmake_minimum_required(VERSION 3.16)
  project(TetraMeshTools VERSION 1.0)
  
  list(APPEND CMAKE_PREFIX_PATH "/opt/sofa_git/build/lib/cmake")
  list(APPEND CMAKE_PREFIX_PATH "/opt/sofa_git/build/install/lib/cmake")
  
  find_package(Sofa REQUIRED)
  find_library(SofaCore
               NAMES Sofa.Core
               HINTS ${SOFA_ROOT}/lib
  )
  
  if(NOT SofaCore)
    message(FATAL_ERROR "SofaCore library not found")
  endif()
  
  message(${SofaCore})
  
  include_directories(
    ../TetraMeshTools/include
    ../TetraMeshTools/include/TetraMeshTools
    PRIVATE
    include
    ${SOFA_ROOT}/include/Sofa.Config
    ${SOFA_ROOT}/include/Sofa.Helper
    ${SOFA_ROOT}/include/Sofa.Type
    ${SOFA_ROOT}/include/Sofa.Core
    ${SOFA_ROOT}/include/Sofa.Topology
    ${SOFA_ROOT}/include/Sofa.Geometry
    ${SOFA_ROOT}/include/Sofa.DefaultType
    ${SOFA_ROOT}/include/Sofa.LinearAlgebra
    ${SOFA_ROOT}/include/Sofa.Component.Compat
    ${SOFA_ROOT}/include/Sofa.Component.Engine.Generate
  )
  
  add_library(
    SofaTetraStuffing
    SHARED
    src/SofaTetraStuffing.cpp
  )
  
  link_directories(
    ${SOFA_ROOT}/lib
  )
  
  target_link_libraries(
    SofaTetraStuffing
    ${SofaCore}
  )
  
  add_compile_options(-O3 -Wall -Wextra -Wpedantic -fopenmp)
```
Running 
`cmake -B build/ -D CMAKE_BUILD_TYPE=Release`
`cmake --build build/`
in my own project complains that "SofaConfig.cmake" could not be found:
```
-- The C compiler identification is GNU 11.4.0
-- The CXX compiler identification is GNU 11.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:8 (find_package):
  By not providing "FindSofa.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Sofa", but
  CMake did not find one.

  Could not find a package configuration file provided by "Sofa" with any of
  the following names:

    SofaConfig.cmake
    sofa-config.cmake

  Add the installation prefix of "Sofa" to CMAKE_PREFIX_PATH or set
  "Sofa_DIR" to a directory containing one of the above files.  If "Sofa"
  provides a separate development package or SDK, be sure it has been
  installed.
```

So I went looking in $my_sofa_path/build/install/lib/cmake/Sofa.Config and found "Sofa.ConfigConfig.cmake". Funnily enough I could fix this by simply copying `Sofa.ConfigConfig.cmake` to `SofaConfig.cmake` in the Sofa.Config folder.
Does this file have a wrong name? Or is my CMakeLists trying to find the wrong package name?

**Comment by dennis2society** [comment, marked as answer]
Never mind. My fault. My CMakeLists.txt was not correct. Changing it to this now works fine:
```
cmake_minimum_required(VERSION 3.16)
project(TetraMeshTools VERSION 1.0)

# Set this to where you have installed your Sofa build:
set(SOFA_INSTALL_DIR "/opt/sofa_git/build/install/")

# This can/should be done by providing the CMAKE_PREFIX_PATH="my_sofa_path"
list(APPEND CMAKE_PREFIX_PATH ${SOFA_INSTALL_DIR}/lib/cmake)


find_package(Sofa.Config REQUIRED)

find_library(SofaCore
             NAMES Sofa.Core
             HINTS ${SOFA_INSTALL_DIR}/lib
)

if(NOT SofaCore)
  message(FATAL_ERROR "SofaCore library not found")
endif()

message(${SofaCore})

include_directories(
  ../TetraMeshTools/include
  ../TetraMeshTools/include/TetraMeshTools
  PRIVATE
  include
  ${SOFA_INSTALL_DIR}/include/Sofa.Config
  ${SOFA_INSTALL_DIR}/include/Sofa.Helper
  ${SOFA_INSTALL_DIR}/include/Sofa.Type
  ${SOFA_INSTALL_DIR}/include/Sofa.Core
  ${SOFA_INSTALL_DIR}/include/Sofa.Topology
  ${SOFA_INSTALL_DIR}/include/Sofa.Geometry
  ${SOFA_INSTALL_DIR}/include/Sofa.DefaultType
  ${SOFA_INSTALL_DIR}/include/Sofa.LinearAlgebra
  ${SOFA_INSTALL_DIR}/include/Sofa.Component.Compat
  ${SOFA_INSTALL_DIR}/include/Sofa.Component.Engine.Generate
)

add_library(
  SofaTetraStuffing
  SHARED
  src/SofaTetraStuffing.cpp
)

link_directories(
  ${SOFA_INSTALL_DIR}/lib
)

target_link_libraries(
  SofaTetraStuffing
  ${SofaCore}
)

add_compile_options(-O3 -Wall -Wextra -Wpedantic -fopenmp)
```
